addons/api/addon/services/indexed-db.js

```diff
@@ -48,17 +51,31 @@ const convertBooleansInObject = (result, object) => {
 };
 
 /**
- * Recursively copy over the object and convert any string "true" or "false" back to booleans
+ * Recursively copy over the object and convert any string "true" or "false" back to booleans.
+ * Will also convert any array fields back using their model transformation.
  * @param result
  * @param object
+ * @param schema
+ * @param serializer
  * @returns {*}
  */
-const unconvertBooleansInObject = (result, object) => {
+const unconvertFields = (result, object, schema, serializer) => {
   Object.entries(object).forEach(([key, value]) => {
     if (typeOf(value) === 'object') {
-      result[key] = unconvertBooleansInObject({}, value);
+      result[key] = unconvertFields({}, value, schema, serializer);
     } else if (value === 'true' || value === 'false') {
       result[key] = value === 'true';
+    } else if (typeOf(value) === 'array') {
+      const attribute = schema.attributes.get(key);
+
+      // If attribute type is array, we transform it back to a tracked array
+      if (attribute?.type === 'array') {
+        const transformer = serializer.transformFor(attribute.type);
+        result[key] = transformer.deserialize(value, attribute.options);
+        return;
+      }
+
+      result[key] = value;
```

Comment:
I could have also just made _any_ array into a tracked array which shouldn't hurt anything but is also not the exact data we would be getting back after a normal normalization.

I also didn't want to just apply a transform deserialization for any array-like object since we have a few fields that are standard arrays but just do simple transforms (e.g `string-array`, `object-array` etc.) and I would be then deserializing twice.

So unfortunately the safest way of handling this is for me to explicitly check for the new `array` type which is the only ember-like array we transform currently.

Reply:
I think this is a good approach to this issue.

Reply:
Agreed, I also think this is the best approach.